tests/pessimistic/pipeline/pipeline.go

```diff
@@ -0,0 +1,329 @@
+	if c.LocalMode {
+		c.dbAddr = "127.0.0.1:4000"
+		c.dbStatusAddr = "127.0.0.1:10080"
+		c.pdAddr = "127.0.0.1:2379"
+	} else {
+		c.dbAddr = dbNode.Address()
+		c.dbStatusAddr = fmt.Sprintf("%s-tidb.%s.svc:10080", dbNode.ClusterName, dbNode.Namespace)
+		c.pdAddr = fmt.Sprintf("%s-pd.%s.svc:2379", pdNode.ClusterName, pdNode.Namespace)
+	}
+}
+
+// SetUp sets up the client.
+func (c *pipelineClient) SetUp(ctx context.Context, nodes []cluster.Node, clientNodes []cluster.ClientNode, idx int) error {
+	if idx != 0 {
+		return nil
+	}
+
+	c.setUpAddrs(clientNodes[0], nodes[0])
+
+	if err := c.openDB(ctx); err != nil {
+		return errors.Trace(err)
+
+	}
+	util.MustExec(c.db, `set @@global.tidb_txn_mode='pessimistic'`)
+	util.MustExec(c.db, `DROP TABLE IF EXISTS `+tableName)
+	util.MustExec(c.db, fmt.Sprintf(`CREATE TABLE %s (id INT PRIMARY KEY, v BIGINT)`, tableName))
+	// Each transaction modifies two rows.
+	for i := 0; i < c.TableSize*2; i++ {
+		util.MustExec(c.db, fmt.Sprintf(`insert into %s values (%d, 0)`, tableName, i))
+	}
+
+	startKey, _, err := c.getTableRange(dbName, tableName)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	c.scheduler = nemesis.NewLeaderShuffler(string(startKey))
+	c.scheduler.SetPDAddr("http://" + c.pdAddr)
+	return nil
+}
+
+func (c *pipelineClient) getTableRange(db, table string) (startKey, endKey []byte, err error) {
+	url := fmt.Sprintf("http://%s/schema/%s/%s", c.dbStatusAddr, db, table)
+	resp, err := httputil.NewHTTPClient(http.DefaultClient).Get(url)
+	if err != nil {
+		return
+	}
+	var body struct {
+		ID int64 `json:"id"`
+	}
+	err = json.Unmarshal(resp, &body)
+	if err != nil {
+		return
+	}
+	startKey, endKey = tablecodec.GetTableHandleKeyRange(body.ID)
+	startKey = codec.EncodeBytes([]byte{}, startKey)
+	endKey = codec.EncodeBytes([]byte{}, endKey)
+	return
+}
+
+// TearDown tears down the client.
+func (c *pipelineClient) TearDown(ctx context.Context, nodes []cluster.ClientNode, idx int) error {
+	c.db.ExecContext(ctx, `DROP DATABASE IF EXISTS `+dbName)
+	return c.db.Close()
+}
+
+// Invoke invokes a request to the database.
+// Mostly, the return Response should implement UnknownResponse interface
+func (c *pipelineClient) Invoke(ctx context.Context, node cluster.ClientNode, r interface{}) core.UnknownResponse {
+	panic("not implemented")
+}
+
+// NextRequest generates a request for latter Invoke.
+func (c *pipelineClient) NextRequest() interface{} {
+	panic("not implemented")
+}
+
+// DumpState the database state(also the model's state)
+func (c *pipelineClient) DumpState(ctx context.Context) (interface{}, error) {
+	panic("not implemented")
+}
+
+// Start runs self scheduled cases
+// this function will block Invoke trigger
+// if you want to schedule cases by yourself, use this function only
+func (c *pipelineClient) Start(ctx context.Context, cfg interface{}, clientNodes []cluster.ClientNode) error {
+	c.scatterTable(dbName, tableName)
```

Comment:
To my understanding, the table `t` only has one region. Will the scatter table works as we expect?

Reply:
Because of the leader shuffler, it will have more than one regions.

Reply:
The table contains 3 regions when running 5 minutes.
```
+-----------+---------------------------+-----------+-----------+-----------------+---------------+------------+---------------+------------+----------------------+------------------+
| REGION_ID | START_KEY                 | END_KEY   | LEADER_ID | LEADER_STORE_ID | PEERS         | SCATTERING | WRITTEN_BYTES | READ_BYTES | APPROXIMATE_SIZE(MB) | APPROXIMATE_KEYS |
+-----------+---------------------------+-----------+-----------+-----------------+---------------+------------+---------------+------------+----------------------+------------------+
| 463       | t_56_5f728000000000000048 | t_59_r_0  | 737       | 5               | 622, 715, 737 | 0          | 269           | 0          | 2                    | 40960            |
| 440       | t_59_r_0                  | t_59_r_34 | 624       | 5               | 477, 624, 673 | 0          | 0             | 0          | 11                   | 164822           |
| 39        | t_59_r_34                 |           | 546       | 5               | 546, 703, 751 | 0          | 4395          | 0          | 35                   | 354064           |
+-----------+---------------------------+-----------+-----------+-----------------+---------------+------------+---------------+------------+----------------------+------------------+
```